src/core.jl

```diff
@@ -16,8 +17,15 @@ const jdouble = Cdouble
 const jsize = jint
 jprimitive = Union{jboolean, jchar, jshort, jfloat, jdouble, jint, jlong}
 
-struct JavaMetaClass{T}
+mutable struct JavaMetaClass{T}
```

Comment:
Why is it necessary to change this to a mutable struct? I haven't look very hard for the answer, but it might save time if someone could explain it.

Reply:
Might it make sense to have mutable and unmutable versions of JavaMetaClass?

Reply:
I'll look around -- I'm pretty sure I had a good reason for that. But maybe not :)

Reply:
Because Java can garbage collect classes and the finalizer on line 84 nulls out ptr for safety. If we can guarantee that deleteref will only ever be called from finalize, it can stay immutable I think.